xbmc/dbwrappers/sqlitedataset.cpp

```diff
@@ -351,6 +351,8 @@ int SqliteDatabase::connect(bool create)
       return DB_CONNECTION_OK;
     }
 
+    sqlite3_close(conn);
```

Comment:
Should this also be in the catch block and/or at the end of the function before the return?

Reply:
@fuzzard, @lrusak, you are right, sqlite3_close() should be extended to this other case, and the sqlite documentation seems to be accurate.